src/Graph.cpp

```diff
@@ -0,0 +1,123 @@
+#include "Graph.hpp"
+
+#include <boost/graph/adjacency_list.hpp>
+#include <unordered_map>
+#include <queue>
+#include <vector>
+#include <string>
+#include <tuple>
+#include <iostream>
+
+namespace booty {
+
+using BoostGraph = boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS>;
+using Vertex = boost::graph_traits<BoostGraph>::vertex_descriptor;
+
+class Graph::GraphImpl {
+public:
+    BoostGraph graph;
+    std::unordered_map<std::string, Vertex> vertexMap;
+    std::unordered_map<Vertex, std::string> labelMap;
+};
+
+Graph::Graph() : impl(new GraphImpl) {}
+
+Graph::~Graph() { delete impl; }
+
+bool Graph::add_node(const std::string& label) {
+    if (impl->vertexMap.find(label) == impl->vertexMap.end()) {
+        Vertex v = boost::add_vertex(impl->graph);
+        impl->vertexMap[label] = v;
+        impl->labelMap[v] = label;
+        return true;
+    }
+    return false;
+}
+
+bool Graph::add_edge(const std::string& from, const std::string& to) {
+    add_node(from);
+    add_node(to);
+    // std::pair<boost::edge_descriptor, bool> is boost::add_edge return type
+    auto result = boost::add_edge(impl->vertexMap[from], impl->vertexMap[to], impl->graph);
+    return result.second; // true if the edge was added, false if it already existed
+}
+
+std::vector<std::string> Graph::bfs(const std::string& startLabel) {
```

Comment:
i am confused here... why did you implement bfs yourslef (kudos, but why?)

https://www.boost.org/doc/libs/1_39_0/libs/graph/doc/breadth_first_search.html